xbmc/pvr/channels/PVRChannelGroup.cpp

```diff
@@ -469,11 +475,71 @@ int CPVRChannelGroup::LoadFromDb()
   if (!database)
     return -1;
 
-  int iChannelCount = Size();
+  const int iChannelCount = Size();
+  const std::vector<std::shared_ptr<CPVRChannelGroupMember>> results = database->Get(*this);
+
+  std::vector<std::shared_ptr<CPVRChannelGroupMember>> membersToDelete;
+  if (!results.empty())
+  {
+    CSingleLock lock(m_critSection);
+    for (const auto& member : results)
+    {
+      // Consistency check.
+      if (member->ClientID() > 0 && member->ChannelUID() > 0 && member->IsRadio() == IsRadio())
+      {
+        m_sortedMembers.emplace_back(member);
+        m_members.insert(
+            std::make_pair(std::make_pair(member->ClientID(), member->ChannelUID()), member));
```

Comment:
this is safe to emplace
```suggestion
        m_members.emplace(std::make_pair(member->ClientID(), member->ChannelUID()), member);
```

Reply:
Cool, thanks.

Reply:
I changed all occurrences of `m_members.insert`